src/main/java/mate/academy/service/impl/UserServiceImpl.java

```diff
@@ -2,26 +2,39 @@
 
 import java.util.Optional;
 import mate.academy.dao.UserDao;
+import mate.academy.exception.RegistrationException;
 import mate.academy.lib.Inject;
 import mate.academy.lib.Service;
 import mate.academy.model.User;
+import mate.academy.service.ShoppingCartService;
 import mate.academy.service.UserService;
 import mate.academy.util.HashUtil;
 
 @Service
 public class UserServiceImpl implements UserService {
     @Inject
     private UserDao userDao;
+    @Inject
+    private ShoppingCartService shoppingCartService;
 
     @Override
     public User add(User user) {
+        String currentPassword = user.getPassword();
         user.setSalt(HashUtil.getSalt());
         user.setPassword(HashUtil.hashPassword(user.getPassword(), user.getSalt()));
-        return userDao.add(user);
+        if (user.getPassword().equals(currentPassword)) {
+            throw new RegistrationException("password was not encrypted");
+        }
+        userDao.add(user);
+        shoppingCartService.registerNewShoppingCart(user);
+
+        return userDao.get(user.getEmail()).orElseThrow(
+                () -> new RegistrationException(
+                        "registration failed, " + user + " not in database"));
```

Comment:
Consider using a more specific exception type or message for better error handling and clarity when the user is not found in the database after registration.